cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -671,6 +678,22 @@ private void emitProvisioningTimeMetric(ProgramRunId programRunId, ProfileId pro
                                                     provisioningTime);
   }
 
+  /**
+   * Emit the pipeline run time metric. The tags are constructed with the program run id and program run status.
+   */
+  private void emitRunTimeMetric(ProgramRunId programRunId, ProgramRunStatus programRunStatus,
+                                 long runTime) {
+    Map<String, String> tags = ImmutableMap.<String, String>builder()
+      .put(Constants.Metrics.Tag.NAMESPACE, programRunId.getNamespace())
+      .put(Constants.Metrics.Tag.APP, programRunId.getApplication())
+      .put(Constants.Metrics.Tag.COMPONENT, Constants.Service.APP_FABRIC_HTTP)
```

Comment:
Why we need component? Component is not for user program context. Also, RunId should be in the context, right? See https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/AbstractContext.java#L290 for how a program run context should be